igvc_description/urdf/jessi.urdf.xacro

```diff
@@ -0,0 +1,460 @@
+                <far>100</far>
+              </clip>
+              <!--
+                  <distortion>
+                  <k1>-0.25</k1>
+                  <k2>0.12</k2>
+                  <k3>0.0</k3>
+                  <p1>-0.00028</p1>
+                  <p2>-0.00005</p2>
+                  <center>0.5 0.5</center>
+                  </distortion>
+              -->
+          </camera>
+          <always_on>1</always_on>
+          <update_rate>30</update_rate>
+          <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
+              <alwaysOn>true</alwaysOn>
+              <updateRate>30</updateRate>
+              <cameraName>center_cam</cameraName>
+              <imageTopicName>image_raw</imageTopicName>
+              <cameraInfoTopicName>camera_info</cameraInfoTopicName>
+              <frameName>optical_center_cam</frameName>
+              <hackBaseline>0.0</hackBaseline>
+              <distortionK1>0.0</distortionK1>
+              <distortionK2>0.0</distortionK2>
+              <distortionK3>0.0</distortionK3>
+              <distortionT1>0.0</distortionT1>
+              <distortionT2>0.0</distortionT2>
+          </plugin>
+      </sensor>
+  </gazebo>
+
+
+    <!-- base of the robot, should be ground level directly below center points of wheels -->
+  <link name="base_link">
+    <xacro:null_link/>
+  </link>
+  <gazebo reference="base_link">
+    <material>Gazebo/Grey</material>
+  </gazebo>
+
+  <link name="base_footprint">
+    <xacro:null_link/>
+  </link>
+  <gazebo reference="base_link">
+    <material>Gazebo/Grey</material>
+  </gazebo>
+
+  <joint name="base_footprint_to_base_link" type="fixed">
+    <parent link="base_footprint"/>
+    <child link="base_link"/>
+    <origin xyz="0.0 0.0 ${drive_tire_diameter / 2}" rpy="0 0 0"/>
+    <axis xyz="0 0 0"/>
+  </joint>
+
+  <gazebo reference="body">
+    <material>Gazebo/Grey</material>
+    <mu1>0.2</mu1>
+    <mu2>0.2</mu2>
+  </gazebo>
+
+  <!-- TODO parameterize -->
+  <!-- collision range = min sensing distance -->
+  <!-- gpu does gpu acceleration -->
+  <VLP-16 parent="base_link" name="lidar" topic="/scan/pointcloud" hz="10" samples="500" gpu="false">
+    <origin xyz="${lidar_forward} 0.0 ${lidar_height}" rpy="0 ${lidar_pitch} 0" />
+  </VLP-16>
+
+  <xacro:wheel lr_prefix="left" lr_reflect="1" tire_diameter="${drive_tire_diameter}"
+               wheel_mass="2.0" axle_eff_limit="10" wheel_width="${drive_tire_width}"/>
+  <xacro:wheel lr_prefix="right" lr_reflect="-1" tire_diameter="${drive_tire_diameter}"
+               wheel_mass="2.0" axle_eff_limit="10" wheel_width="${drive_tire_width}"/>
+
+  <xacro:caster ball_radius="${caster_radius}" ball_mass="1.0"
+                tire_diameter="${drive_tire_diameter}" x="${caster_forward}" y="0"/>
+
+
+  <link name="body">
+      <inertial>
+          <mass value="20" />
+          <origin xyz="0 0 0" rpy="0 0 0"/>
+          <inertia  ixx="${2 / 5 * 20 * 1 ** 2}" ixy="0.0" ixz="0.0"
```

Comment:
Where does this equation come from? Is it intentionally left spelled out here instead of just putting the numbers in directly? Is the order of evaluation known in this context?